src/languages/PlSqlFormatter.js

```diff
@@ -0,0 +1,113 @@
+    "NAME", "NAN", "NATIONAL", "NATIVE", "NATURAL", "NATURALN", "NCHAR", "NEW", "NEXTVAL", "NOCOMPRESS", "NOCOPY", "NOT",
+    "NOWAIT", "NULL", "NULLIF", "NUMBER_BASE", "NUMBER",
+    "OBJECT", "OCICOLL", "OCIDATE", "OCIDATETIME", "OCIDURATION", "OCIINTERVAL", "OCILOBLOCATOR", "OCINUMBER", "OCIRAW",
+    "OCIREF", "OCIREFCURSOR", "OCIROWID", "OCISTRING", "OCITYPE", "OF", "OLD", "ON", "ONLY", "OPAQUE", "OPEN", "OPERATOR",
+    "OPTION", "ORACLE", "ORADATA", "ORDER", "ORGANIZATION", "ORLANY", "ORLVARY", "OTHERS", "OUT", "OVERLAPS",
+    "OVERRIDING",
+    "PACKAGE", "PARALLEL_ENABLE", "PARAMETER", "PARAMETERS", "PARENT", "PARTITION", "PASCAL", "PCTFREE", "PIPE", "PIPELINED",
+    "PLS_INTEGER", "PLUGGABLE", "POSITIVE", "POSITIVEN", "PRAGMA", "PRECISION", "PRIOR", "PRIVATE", "PROCEDURE", "PUBLIC",
+    "RAISE", "RANGE", "RAW", "READ", "REAL", "RECORD", "REF", "REFERENCE", "RELEASE", "RELIES_ON", "REM", "REMAINDER",
+    "RENAME", "RESOURCE", "RESULT_CACHE", "RESULT", "RETURN", "RETURNING", "REVERSE", "REVOKE", "ROLLBACK", "ROW", "ROWID",
+    "ROWNUM", "ROWTYPE",
+    "SAMPLE", "SAVE", "SAVEPOINT", "SB1", "SB2", "SB4", "SECOND", "SEGMENT", "SELF", "SEPARATE", "SEQUENCE",
+    "SERIALIZABLE", "SHARE", "SHORT", "SIZE_T", "SIZE", "SMALLINT", "SOME", "SPACE", "SPARSE", "SQL", "SQLCODE",
+    "SQLDATA", "SQLERRM", "SQLNAME", "SQLSTATE", "STANDARD", "START", "STATIC", "STDDEV", "STORED", "STRING", "STRUCT",
+    "STYLE", "SUBMULTISET", "SUBPARTITION", "SUBSTITUTABLE", "SUBTYPE", "SUCCESSFUL", "SUM", "SYNONYM", "SYSDATE",
+    "TABAUTH", "TABLE", "TDO", "THE", "THEN", "TIME", "TIMESTAMP", "TIMEZONE_ABBR", "TIMEZONE_HOUR", "TIMEZONE_MINUTE",
+    "TIMEZONE_REGION", "TO", "TRAILING", "TRANSACTION", "TRANSACTIONAL", "TRIGGER", "TRUE", "TRUSTED", "TYPE",
+    "UB1", "UB2", "UB4", "UID", "UNDER", "UNIQUE", "UNPLUG", "UNSIGNED", "UNTRUSTED", "USE", "USER", "USING",
+    "VALIDATE", "VALIST", "VALUE", "VARCHAR", "VARCHAR2", "VARIABLE", "VARIANCE", "VARRAY", "VARYING", "VIEW", "VIEWS", "VOID",
+    "WHENEVER", "WHILE", "WITH", "WORK", "WRAPPED", "WRITE",
+    "YEAR",
+    "ZONE",
+];
+
+const reservedToplevelWords = [
+    "ADD", "ALTER COLUMN", "ALTER TABLE",
+    "BEGIN",
+    "CONNECT BY",
+    "DECLARE", "DELETE FROM", "DELETE",
+    "END", "EXCEPT", "EXCEPTION",
+    "FETCH FIRST", "FROM",
+    "GROUP BY",
+    "HAVING",
+    "INSERT INTO", "INSERT", "INTERSECT",
+    "LIMIT", "LOOP",
+    "MODIFY",
+    "ORDER BY",
+    "SELECT", "SET CURRENT SCHEMA", "SET SCHEMA", "SET", "START WITH",
+    "UNION ALL", "UNION", "UPDATE",
+    "VALUES",
+    "WHERE"
+];
+
+const reservedNewlineWords = [
+    "AND",
+    "CROSS APPLY", "CROSS JOIN",
+    "ELSE", "END",
+    "INNER JOIN",
+    "JOIN",
+    "LEFT JOIN", "LEFT OUTER JOIN",
+    "OR", "OUTER APPLY", "OUTER JOIN",
+    "RIGHT JOIN", "RIGHT OUTER JOIN",
+    "WHEN",
+    "XOR"
+];
+
+let tokenizer;
+
+export default class PlSqlFormatter {
```

Comment:
Went over the 2 lists:
release 1 (12.1) - https://docs.oracle.com/database/121/LNPLS/reservedwords.htm#LNPLS019
release 2 (9.2) - https://docs.oracle.com/cd/A97630_01/appdev.920/a96624/f_words.htm

I can see that the list you have is combination of both with these additional fields:
`[
    "NEXTVAL",
    "CROSS APPLY",
    "CROSS JOIN",
    "INNER JOIN",
    "JOIN",
    "LEFT JOIN",
    "LEFT OUTER JOIN",
    "OUTER APPLY",
    "OUTER JOIN",
    "RIGHT JOIN",
    "RIGHT OUTER JOIN",
    "XOR",
    "ALTER COLUMN",
    "ALTER TABLE",
    "CONNECT BY",
    "DELETE FROM",
    "FETCH FIRST",
    "GROUP BY",
    "INSERT INTO",
    "ORDER BY",
    "SET CURRENT SCHEMA",
    "SET SCHEMA",
    "START WITH",
    "UNION ALL"
]`

I also see that some of the fields are only defined in https://docs.oracle.com/cd/A97630_01/appdev.920/a96624/f_words.htm and not https://docs.oracle.com/database/121/LNPLS/reservedwords.htm#LNPLS019

`[
    "BINARY_INTEGER",
    "BOOLEAN",
    "COALESCE",
    "CURRVAL",
    "DECIMAL",
    "DO",
    "EXTENDS",
    "EXTRACT",
    "FALSE",
    "INTEGER",
    "MLSLABEL",
    "NATURAL",
    "NATURALN",
    "NULLIF",
    "NUMBER",
    "PCTFREE",
    "PLS_INTEGER",
    "POSITIVE",
    "POSITIVEN",
    "REAL",
    "RELEASE",
    "ROWID",
    "ROWNUM",
    "ROWTYPE",
    "SMALLINT",
    "SPACE",
    "SQLERRM",
    "SUCCESSFUL",
    "SYSDATE",
    "TRIGGER",
    "TRUE",
    "UID",
    "USER",
    "VALIDATE",
    "VARCHAR",
    "VARCHAR2",
    "WHENEVER",
]`

Not sure which documentation has the correct data or maybe combination of both is the right way?